blog/content/post/2021-02-21-amd-milan-osf.md

```diff
@@ -0,0 +1,153 @@
+           firmware ecosystem by providing silicon initialization code. With
+           a few years break AMD is now trying to support open source firmware
+           again with the mobile platform like Chromebooks. However, the recent
+           achievements have made everybody think that open source firmware is
+           also possible on severs.'
+cover: /covers/amd-epyc.jpg
+author: michal.zygowski
+layout: post
+published: true
+date: 2021-02-21
+archives: "2021"
+
+tags:
+  - firmware
+categories:
+  - Firmware
+
+---
+
+# Introduction
+
+Do you remember times when AMD was actively supporting open source firmware?
+Back in the 2008 AMD was delivering the AGESA silicon initialization code and
+was contributing it to coreboot. Up to 2014 AMD was release the AGESA code in
+public. These were the golden years of open source firmware for AMD processors
+and many platforms emerged in coreboot. You may know those platform under
+various names like PC Engines ALIX and APU1. They benefit still from fully open
+source implementation. However since 2014 AMD started to deliver the AGESA in
+binary form only. And that situation is maintained till today with some
+exceptions (after AMD Steppe Eagle there were no AGESA binary releases
+anymore). Google encouraged the AMD to work on FSP compatible interface for
+AGESA for their Chromebooks and today we have a AMD Picasso support in
+coreboot. If you want to know more about Open Source Firmware on AMD history,
+please refer to our FOSDEM presentations:
+
+* https://video.fosdem.org/2020/K.4.401/coreboot_amd.mp4
+* https://video.fosdem.org/2021/D.firmware/firmware_osfsoap2.mp4
+
+Everybody hopes that AMD server will get open source firmware too. According to
+rumors something is moving into that direction. However, in this blog post I
+will describe the real current situation.
+
+## OSF on AMD Milan
+
+Since the Ron Minnich talk about [pure open source on an AMD Zen](https://vimeo.com/488147337)
+we have been getting more requests for open source firmware on AMD based
+servers like Rome or Milan. While this isn't surprising, many people seem to be
+missing a few important points and get the wrong idea (by the way big kudos to
+Ron for attempting and showing it on OSFC2020). The number of issues
+encountered by Ron is enormous (and not all has been shown) and typically
+related to very basic thing like timers. Only those tightly related to firmware
+development can understand that this is only the beginning of the road full of
+obstacles. There are many more undocumented registers that are set by AGESA and
+many more issues to find out yet. One may forget to use advanced features
+without AGESA. One more important thing is that the presentation noted that SMP
+is not yet well understood. That means only one main core is running, while the
+rest is not initialized, where on server platforms is basically a killer
+performance limitation.
```

Comment:
@miczyg1 can you refere to specific slides or time pointers in presentation?

Reply:
For some reason, I cannot find it now...